Tests/OpenAITests/OpenAITestsDecoder.swift

```diff
@@ -0,0 +1,205 @@
+//
+//  OpenAITestsDecoder.swift
+//  
+//
+//  Created by Aled Samuel on 10/04/2023.
+//
+
+import XCTest
+@testable import OpenAI
+
+@available(iOS 13.0, *)
+@available(watchOS 6.0, *)
+@available(tvOS 13.0, *)
+class OpenAITestsDecoder: XCTestCase {
+    
+    private let decoder = JSONDecoder()
+    
+    override func setUp() {
+        super.setUp()
+    }
+    
+    func testCompletions() async throws {
```

Comment:
What do you think about some simplification by moving common code to the function: 

```swift
class OpenAITestsDecoder: XCTestCase {

    private func decode<T: Decodable>(_ jsonString: String, _ expectedValue: T) throws {
        let decoder = JSONDecoder()
        let data = jsonString.data(using: .utf8)!
        let decoded = try decoder.decode(T.self, from: data)
        XCTAssertEqual(decoded, expectedValue)
    }

    func testCompletions() throws {
        let data = """
        {
          "id": "foo",
          "object": "text_completion",
          "created": 1589478378,
          "model": "text-davinci-003",
          "choices": [
            {
              "text": "Hello, world!",
              "index": 0,
              "logprobs": null,
              "finish_reason": "length"
            }
          ],
          "usage": {
            "prompt_tokens": 5,
            "completion_tokens": 7,
            "total_tokens": 12
          }
        }
        """
        let expectedValue = CompletionsResult(id: "foo", object: "text_completion", created: 1589478378, model: .textDavinci_003, choices: [.init(text: "Hello, world!", index: 0)], usage: .init(promptTokens: 5, completionTokens: 7, totalTokens: 12))
        try decode(data, expectedValue)
    }

   //Other tests here
}
```